.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,51 @@
+name: Docker
+
+on:
+  push:
+    branches:
+      - master
+      - Dockerfile-use-installer
+  pull_request:
+    branches:
+      - master
+      - Dockerfile-use-installer
+  workflow_dispatch:
+
+env:
+  IMAGE_NAME: fomu-toolchain
+
+jobs:
+  push:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build image
+        run: docker build ./dockerfiles --file dockerfiles/Dockerfile.use-installer --tag $IMAGE_NAME
+
+      - name: Log into registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u ${{ github.actor }} --password-stdin
```

Comment:
Docker provides a GitHub Action which should securely handle the login and ensure logout, regardless of failing steps. See https://www.docker.com/blog/docker-github-actions/ and https://github.com/docker/login-action#github-packages-docker-registry.